smart_card_connector_app/src/chrome-api-provider.js

```diff
@@ -511,20 +575,19 @@ GSC.ConnectorApp.ChromeApiProvider = class extends goog.Disposable {
       goog.log.log(
```

Comment:
These `try`-`catch` blocks seems like something that can be automatically done by some kind of a wrapper as well? Just a thought - I haven't thought it through, and no need to try it in the current PR.

Reply:
i wanted to do smth like that, but i'm not sure how can that be done nicely.
The problem is that if the error is thrown, the wrapper will need to know what are the default values it needs to return.

Reply:
You can pass the default values as another parameter for that helper (`ChromeEventListener` or another helper method).

Ultimately, looking at the repetitive patterns across these methods, it seems that they can be replaced with a single generic method? It just needs to know the things that differ, something like: (1) Chrome API event name, (2) lambda that "massages" input arguments for sending them as PC/SC calls, (3) same for output arguments, (4) report*Result function, (5) default arguments for the report function.